### ISSUE TITLE
A random smattering of the source files use crlf line terminators

### ISSUE THREAD

**Issue by ChrisKitching** [klemens-morgenstern/boost-process#74]
Any particular reason for the inconsistency?

**Comment by klemens-morgenstern** [comment]
Different editors on windows & linux. I looked out to use spaces instead of tabs, but didn't even notice that. Just out of curiousity: where does that become a problem?

**Comment by ChrisKitching** [comment]
Some tools dislike mixed line endings in the same files. 

People using editors which normalise newlines on load (which are annoyingly plentiful) will end up causing spurious git changes (as all those lines count as edited).

Bad things can happen if other people haven't carefully read the git manual page about autocrlf (which, incidentally, is how you use different editors across windows/Linux and avoid ending up in this situation): https://help.github.com/articles/dealing-with-line-endings/

In short: developers have to be aware that this is the situation, and take action to avoid cluttering git history with useless changes as a result.

**Comment by klemens-morgenstern** [comment]
Makes, sense I didn't notice.

**Comment by ChrisKitching** [comment]
Incidentally, your commit only changed lines that consist _only_ of a newline. Now files that used to be pure crlf are a mixture of both, which is in many ways even worse, since the default "save" behaviour of many editors is to set all line endings to the same character.

The amount of lines you'd end up touching to convert everything to `\n` turns out to be enormous: running `dos2unix` over the whole repo ends up with a diffstat of `58 files changed, 6324 insertions(+), 6324 deletions(-)`. Clobbering that amount of `git blame` is arguably not a good idea.

I guess the thing to do might be to make all files use one or the other and consistently use only one going forward?

Another alternative would be to history-rewrite the entire repository, although that's a terrible idea for other reasons.

Certainly, you might want to force-push off that big commit you did...

**Comment by klemens-morgenstern** [comment]
I'll try to figure something out, git is killing me with that on windows.
There is however no style requirement in boost, so that'll be alright for now I guess.
